etc/eslint/rules/typescript.js

```diff
@@ -0,0 +1,2426 @@
+*
+* @example
+* // Bad...
+* module M {
+*     export const x = 1;
+* }
+*
+* // Good...
+* namespace N {
+*     export const x = 1;
+* }
+*/
+rules[ '@typescript-eslint/no-namespace' ] = 'error';
+
+/**
+* Disallows non-null assertions using the `!` postfix operator.
+*
+* @name no-non-null-assertion
+* @memberof rules
+* @type {string}
+* @default 'error'
+* @see [no-non-null-assertion]{@link https://typescript-eslint.io/rules/no-non-null-assertion}
+*
+* @example
+* // Bad...
+* const x: number | null = null;
+* const y: number = x!;
+*
+* // Good...
+* if ( x !== null ) {
+*     const y: number = x;
+* }
+*/
+rules[ '@typescript-eslint/no-non-null-assertion' ] = 'error';
+
+/**
+* Disables the rule against using `require` statements instead of import declarations.
+*
+* @name no-require-imports
+* @memberof rules
+* @type {string}
+* @default 'off'
+* @see [no-require-imports]{@link https://typescript-eslint.io/rules/no-require-imports}
+*
+* @example
+* // Okay...
+* const module = require( 'module' );
+*/
+rules[ '@typescript-eslint/no-require-imports' ] = 'off';
+
+/**
+* Disallows aliases for `this`.
+*
+* @name no-this-alias
+* @memberof rules
+* @type {string}
+* @default 'error'
+* @see [no-this-alias]{@link https://typescript-eslint.io/rules/no-this-alias}
+*
+* @example
+* // Bad...
+* const self = this;
+* const that = this;
+*
+* // Good...
+* function example() {
+*   console.log(this);
+* }
+*/
+rules[ '@typescript-eslint/no-this-alias' ] = 'error';
+
+/**
+* Disables the rule against unnecessary boolean literal comparisons.
+*
+* @name no-unnecessary-boolean-literal-compare
+* @memberof rules
+* @type {string}
+* @default 'off'
+* @see [no-unnecessary-boolean-literal-compare]{@link https://typescript-eslint.io/rules/no-unnecessary-boolean-literal-compare}
+*
+* @example
+* // Okay...
+* const x = true;
+* if ( x === true ) {
+*     // ...
+* }
+*/
+rules[ '@typescript-eslint/no-unnecessary-boolean-literal-compare' ] = 'off';
+
+/**
+* Disallows unnecessary qualifiers (like namespaces or enum names) when the referenced item is accessible without them.
+*
+* @name no-unnecessary-qualifier
+* @memberof rules
+* @type {string}
+* @default 'error'
+* @see [no-unnecessary-qualifier]{@link https://typescript-eslint.io/rules/no-unnecessary-qualifier}
+*
+* @example
+* // Bad...
+* namespace A {
+*     export const B = 1;
+* }
+* const x = A.B;
+*
+* // Good...
+* const B = 1;
+* const x = B;
+*/
+rules[ '@typescript-eslint/no-unnecessary-qualifier' ] = 'error';
+
+/**
+* Disallows unnecessary type arguments in generics where the argument is the default.
+*
+* @name no-unnecessary-type-arguments
+* @memberof rules
+* @type {string}
+* @default 'error'
+* @see [no-unnecessary-type-arguments]{@link https://typescript-eslint.io/rules/no-unnecessary-type-arguments}
+*
+* @example
+* // Bad...
+* const x: Array<number> = [];
+*
+* // Good...
+* const x: number[] = [];
```

Comment:
@Planeshifter Is this not in conflict with https://github.com/stdlib-js/stdlib/pull/1126/files#diff-e34afd8893f1cb8f746b105084f21b243edcdab58bdbea94c70bbaaa7ed583c6R84?

Reply:
See `array-type` above.